extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufReader.java

```diff
@@ -102,15 +102,16 @@ public InputRow next()
 
   List<InputRow> readAsList()
   {
+    ByteBuffer buffer = source.getEntity().getBuffer();
     try {
-      ByteBuffer buffer = source.getEntity().getBuffer();
       List<InputRow> rows = parseMetric(Metric.parseFrom(buffer));
       // Explicitly move the position assuming that all the remaining bytes have been consumed because the protobuf
       // parser does not update the position itself
       buffer.position(buffer.limit());
       return rows;
     }
     catch (InvalidProtocolBufferException e) {
+      buffer.position(buffer.limit());
```

Comment:
Why are we moving the buffer position directly to the end? We can't be sure if the exception is at the last element; it might be at any position. So, Here, we might skip more than one event/row.

Reply:
The buffer has a single kafka message, if there's an exception while parsing the proto message then the entire message should be discarded.

Reply:
buffer.position(buffer.limit()) should go in a finally as it's present in try and catch.